plugins/router/template/router.go

```diff
@@ -144,6 +145,20 @@ func newTemplateRouter(cfg templateRouterCfg) (*templateRouter, error) {
 	return router, nil
 }
 
+func endpointsForAlias(alias ServiceAliasConfig, svc ServiceUnit) []Endpoint {
+	if len(alias.PreferPort) == 0 {
+		return svc.EndpointTable
+	}
+	endpoints := make([]Endpoint, 0, len(svc.EndpointTable))
+	for i := range svc.EndpointTable {
+		endpoint := svc.EndpointTable[i]
+		if endpoint.PortName == alias.PreferPort || endpoint.Port == alias.PreferPort {
```

Comment:
On Sep 10, 2015, at 5:23 PM, Paul Weil <notifications@github.com> wrote:

In plugins/router/template/router.go
<https://github.com/openshift/origin/pull/4606#discussion_r39215815>:

> @@ -144,6 +145,20 @@ func newTemplateRouter(cfg templateRouterCfg) (*templateRouter, error) {
>  	return router, nil
>  }
>
> +func endpointsForAlias(alias ServiceAliasConfig, svc ServiceUnit) []Endpoint {
> +	if len(alias.PreferPort) == 0 {
> +		return svc.EndpointTable
> +	}
> +	endpoints := make([]Endpoint, 0, len(svc.EndpointTable))
> +	for i := range svc.EndpointTable {
> +		endpoint := svc.EndpointTable[i]
> +		if endpoint.PortName == alias.PreferPort || endpoint.Port == alias.PreferPort {

Right, it filters to the preferred port. So if I define multiple ports in
my route it doesn't do anything with the 0+n ports. Just a bit confusing
that I can configure for multiple ports and it ignores them. It doesn't
seem to be a reasonable use case to want 80 and 8888 of your pod on the
same route anyway but if we did want to support it we'd need to plumb
through the route.Ports. Just another reason to get rid of this internal
representation.



Ideally, a sophisticated router would match route ports to features.
However, it's not worth the complexity.  I left multiple ports because we
will have to deal with it in the future (f5)


—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/4606/files#r39215815>.